tokio/src/runtime/thread_pool/worker.rs

```diff
@@ -639,7 +675,17 @@ impl Shared {
             core.run_queue.push_back(task, &self.inject);
             true
         } else {
-            core.run_queue.push(task, &self.inject)
+            // Push to the LIFO slot
+            let prev = core.lifo_slot.take();
+            let ret = prev.is_some();
+
+            if let Some(prev) = prev {
+                core.run_queue.push_back(prev, &self.inject);
+            }
```

Comment:
nitpicky, take it or leave it: seems like this could be 
```suggestion
            let ret = if let Some(prev) = prev {
                core.run_queue.push_back(prev, &self.inject);
                true
            } else {
                false
            }:
```
although...i'm not actually sure if that's more readable...